app.py

```diff
@@ -0,0 +1,341 @@
+            cur.execute(q)
+        return jsonify(status_code=200, response=response)
+    except KeyError as e:
+        return jsonify(
+            message=f"Missing key in payload: {type(e)}",
+            args=e.args,
+            status_code=400,
+            error_type="Bad Request",
+        )
+    except psycopg2.Error as e:
+        return jsonify(
+            message=f"Psycopg2 driver error: {type(e)}",
+            args=e.args,
+            status_code=500,
+            error_type="Internal Server Error",
+        )
+    except Exception as e:
+        return jsonify(
+            message=f"Internal Server Error: {type(e)}",
+            args=e.args,
+            status_code=500,
+            error_type="Internal Server Error",
+        )
+
+
+@app.route("/search_plant_info/<name>", methods=["GET"])
+def search_plant_info(name: str) -> Response:
+
+    """
+    Endpoint fetching plant data from cache or from Trefle API
+
+    curl http://localhost:5000/search_plant_info/tapertip%20onion
+    """
+
+    try:
+        data = get_plant_info(name, 100)
+        return jsonify(status_code=200, data=data)
+    except Exception as e:
+        return jsonify(
+            message=f"Internal Server Error: {type(e)}",
+            args=e.args,
+            status_code=500,
+            error_type="Internal Server Error",
+        )
+
+
+@app.route("/identify_plant", methods=["POST"])
+def identify_plant():
+    """
+    Endpoint receiving an image and querying Plant.id to identify the plant species
+
+    curl -X POST http://localhost:5000/identify_plant -F "file=@./tests/resources/test_flower_img.jpeg" -H "Content-Type: multipart/form-data"
+    """
+
+    if "file" not in request.files:
+        return jsonify(
+            message=f"Bad Request: file missing",
+            status_code=400,
+            error_type="Bad Request",
+        )
+    file = request.files["file"]
+    # if user does not select file, browser also
+    # submit a empty part without filename
+    if file.filename == "" or not file \
+            or not allowed_file(file.filename, ALLOWED_IMG_EXTENSIONS):
+        return jsonify(
+            message=f"Bad Request: file missing",
+            status_code=400,
+            error_type="Bad Request",
+        )
+
+    filename = secure_filename(file.filename)
+    tmp_filename = os.path.join(app.config['UPLOAD_FOLDER'], filename)
+
+    try:
+        logger.info(f"Saving tmp image file at {tmp_filename}")
+        # Save temp file, clean afterwards
+        file.save(tmp_filename)
+
+        # Build plant.id request
+        with open(tmp_filename, "rb") as f:
+            images = [base64.b64encode(f.read()).decode("ascii")]
+        plant_id_payload = {
+            "images": images,
+            "modifiers": ["similar_images"],
+            "plant_details": ["name_authority", "common_names", "url", "wiki_description", "taxonomy"],
```

Comment:
maybe also average height would be cool here as well

Reply:
plant_details is an enum in the plant.id request payload, i'll send you the list of accepted values

Reply:
plant.id api is for the picture identification and returns quite a small amount of data about the plant itself apart from interesting links. i found it quite difficult to gather plant care information even on trefle.io, most of the requests have null values regarding this type of data, I am not sure where to get that info